src/Stratis.Bitcoin.Features.Consensus/Rules/ConsensusRule.cs

```diff
@@ -107,9 +111,26 @@ public class MempoolRuleAttribute : RuleAttribute
     }
 
     /// <summary>
-    /// Rules that provide easy access to the <see cref="PosConsensusRules"/> parent.
+    /// Rules that provide easy access to the <see cref="CoinView"/> which is the store for a PoW system.
+    /// </summary>
+    public abstract class UtxoStoreConsensusRule : ConsensusRule
+    {
+        /// <summary>Allow access to the POS parent.</summary>
+        protected PowConsensusRules PowParent;
+
+        /// <inheritdoc />
+        public override void Initialize()
+        {
+            this.PowParent = this.Parent as PowConsensusRules;
+
+            Guard.NotNull(this.PowParent, nameof(this.PowParent));
+        }
+    }
+
+    /// <summary>
+    /// Rules that provide easy access to the <see cref="StakeChain"/> which is the store for a PoS system.
```

Comment:
Reference cannot be resolved. 
Please replace StakeChain with IStakeChain or StakeChainStore